source/m2mfirmware.cpp

```diff
@@ -430,5 +430,7 @@ bool M2MFirmware::check_value_range(FirmwareResource resource, int64_t value) co
 
 void M2MFirmware::set_zero_value(M2MResource *resource)
 {
-    resource->set_value((const uint8_t*)"0", 1);
+    if(resource) {
+        resource->set_value((const uint8_t*)"0", 1);
+    }
```

Comment:
Don't want the if, probably. Gives the immediate impression of being an incorrect fix. Why would you want to allow NULL inputs to this function?